pkg/client/restclient/client.go

```diff
@@ -119,6 +135,33 @@ func readExpBackoffConfig() BackoffManager {
 			time.Duration(backoffDurationInt)*time.Second)}
 }
 
+// createSerializers creates all necessary serializers for given contentType.
+func createSerializers(config ContentConfig) (*Serializers, error) {
+	negotiated := config.NegotiatedSerializer
+	contentType := config.ContentType
+	serializer, ok := negotiated.SerializerForMediaType(contentType, nil)
+	if !ok {
+		return nil, fmt.Errorf("serializer for %s not registered", contentType)
+	}
+	streamingSerializer, framer, _, ok := negotiated.StreamingSerializerForMediaType(contentType, nil)
+	if !ok {
+		return nil, fmt.Errorf("streaming serializer for %s not registered", contentType)
+	}
+	if framer == nil {
+		return nil, fmt.Errorf("no framer for %s", contentType)
+	}
+	internalGV := unversioned.GroupVersion{
+		Group:   config.GroupVersion.Group,
+		Version: runtime.APIVersionInternal,
```

Comment:
The clientset clients should deocode (actually just deserialize) to versioned objects, so for them the version here should be config.GroupVersion.Version. How about adding another field to the client config to specify the decode version? I can send a PR if you think adding an extra config field is acceptable. @wojtek-t @smarterclayton